test/test_torch.py

```diff
@@ -2388,8 +2394,11 @@ def test_index_add(self):
     def test_index_add_all_dtypes(self):
         for device in torch.testing.get_all_device_types():
             for dtype in torch.testing.get_all_math_dtypes(device):
+                # index_add is not supported for complex dtypes on cuda yet
+                if device.startswith('cuda') and dtype.is_complex:
```

Comment:
Yick. I wonder why `get_all_device_types` doesn't give us `torch.device` objects as opposed to strings.